### DISCUSSION TITLE
EnvironmentFunctionalEvaluationError in Prisma Introspect

### DISCUSSION THREAD

**Discussion by Redzli** [prisma/prisma#2818] [Q&A] [answered]
Hi, I am having a problem that cannot be solved by installing @prisma/cli@dev

I have a simple Postgres db setup with the following schema:

```sql
CREATE TABLE public.users (
  id SERIAL PRIMARY KEY NOT NULL,
  email TEXT NOT NULL,
  display_name TEXT NOT NULL,
  is_admin BOOLEAN NOT NULL DEFAULT FALSE,
  created_at TIMESTAMP DEFAULT NOW(),
  updated_at TIMESTAMP DEFAULT NOW()
);
```

I then ran `yarn prisma introspect --schema ./db/schema.prisma` and received error:

```bash
Introspecting based on datasource defined in db/schema.prisma …
Error: Error in datamodel: ErrorCollection { errors: [EnvironmentFunctionalEvaluationError { var_name: "DATABASE_URL", span: Span { start: 109, end: 128 } }] }
```
I have my DATABASE_URL injected as
`DATABASE_URL=postgresql://postgres:postgres@localhost:5433/project_dev`

I am also using a docker image for the Postgres instance:

```
docker run -p "5433:5432" -e POSTGRES_USER=<user> s -e POSTGRES_PASSWORD=<password> -e POSTGRES_HOSTNAME=localhost -e POSTGRES_DB=<DB> postgres:9.6
```

I am using `"@prisma/cli": "^2.1.0-dev.52",`

Can you let me know how to resolve it? Many thanks in advance

**Comment by ryands17** [comment]
Hey @Redzli :wave: 
The DB url has port `5433` but you're Postgres image has port `5432`. I think that might be causing the issue.

**Reply by Redzli** [reply]
Hi, thanks for the reply. The reason for `-p "5433:5432"` is to bind my localhost port (5433) to container port (5432). It looks like they are consistent.

**Reply by ryands17** [reply]
@Redzli Could you share in what way are you injecting the database url environment variable?

**Reply by Redzli** [reply]
I am building a `nextjs` app so I ran
```
source .env.development.local
```

 Can confirm the env variable is injected by running `echo $DATABASE_URL`

**Comment by janpio** [comment]
Can you try to put the schema in the standard path `./prisma/schema.prisma` and see if the error persists?

**Reply by Redzli** [reply]
Hi, thanks for your reply. I don't have the authorisation to change the path because several of my colleagues are collaborating on the project. Would there be any other workaround?

**Reply by janpio** [reply]
This would not be a permanent change, but you allow us to understand of the location of the schema is the root cause for this problem. You can just try it out locally, and then report back if it solves the problem or not.

**Reply by Redzli** [reply, marked as answer]
Thanks. I ran two introspections, one at `./db/schema.prisma`, the other at `./prisma/schema.prisma`. Both introspections didn't reproduce the error. Seems the problem is resolved. On another note, can you let me know what possibly resulted in the bug?

**Reply by janpio** [reply]
My theory was that the schema location influenced the problem in some way - but this seems not to be true if changing the schema location did not do anything, but it just went away by itself 🤷

**Comment by ludralph** [comment]
Hi @Redzli,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
